### PR TITLE
Disable THP in check-before-deployment.md

### PR DESCRIPTION
**Pull request by pepezzzz**
add Disabling Transparent HugePages section

### What is changed, added or deleted? (Required)

Disabling transparent hugepages before TiDB installation.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by TomShawn** [comment]
@pepezzzz Only dev and v4.0 have this `check-before-deployment.md` document, and I'll update the version box section in the description. 

Also, please involve a technical review.

File: check-before-deployment.md [hunk 1]
```diff
@@ -229,6 +229,103 @@ sudo systemctl start ntpd.service && \
...
+    ```
+    /boot/vmlinuz-3.10.0-957.el7.x86_64
+    ```
+
+3. 执行 `grubby --update-kernel` 命令修改内核配置：
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    grubby --args="transparent_hugepage=never" --update-kernel /boot/vmlinuz-3.10.0-957.el7.x86_64
```
Comment (ethercflow) [inline check-before-deployment.md:271]: Notice: Certain ktune and/or tuned profiles specify to enable THP when they are applied.

File: check-before-deployment.md [hunk 2]
```diff
@@ -229,6 +229,103 @@ sudo systemctl start ntpd.service && \
 sudo systemctl enable ntpd.service
 ```
 
+## 检测及关闭透明大页
+
+对于数据库应用，不推荐使用 THP（transparent hugepage，透明大页），因为数据库往往具有稀疏而不是连续的内存访问模式，且当高阶内存碎片化比较严重时，分配 THP 页面会出现较大的延迟。若开启针对 THP 的直接内存规整功能，也会出现系统 CPU 使用率激增的现象，因此建议关闭 THP。
+
+采用如下步骤检查是否已经关闭透明大页：
```
Comment (kissmydb) [inline check-before-deployment.md:236]:
```suggestion
采用如下步骤检查是否已经关闭透明大页，并进行关闭：
```

File: check-before-deployment.md [hunk 3]
```diff
@@ -229,6 +229,103 @@ sudo systemctl start ntpd.service && \
...
+5. 执行 `reboot` 命令进行重启或者修改当前的内核配置：
+
+    - 如果需要重启验证，执行 `reboot` 命令：
+       {{< copyable "shell-regular" >}}
+
+        ```bash
+        reboot
+        ```
+
+    - 如果需要修改当前的内核配置：
```
Comment (kissmydb) [inline check-before-deployment.md:308]:
```suggestion
    - 如果不希望重启机器，也可以修改当前的内核配置来立即生效：
```

File: check-before-deployment.md [hunk 4]
```diff
@@ -229,6 +229,103 @@ sudo systemctl start ntpd.service && \
 sudo systemctl enable ntpd.service
 ```
 
+## 检测及关闭透明大页
+
+对于数据库应用，不推荐使用 THP（transparent hugepage，透明大页），因为数据库往往具有稀疏而不是连续的内存访问模式，且当高阶内存碎片化比较严重时，分配 THP 页面会出现较大的延迟。若开启针对 THP 的直接内存规整功能，也会出现系统 CPU 使用率激增的现象，因此建议关闭 THP。
+
+采用如下步骤检查是否已经关闭透明大页，并进行关闭：
+
+1. 执行以下命令，如果输出 `[always] madvise never` 表示透明大页处于启用状态：
```
Comment (TomShawn) [inline check-before-deployment.md:238]:
```suggestion
1. 执行以下命令查看透明大页的开启状态。如果返回 `[always] madvise never` 则表示处于启用状态：
```

File: check-before-deployment.md [hunk 5]
```diff
@@ -229,6 +229,103 @@ sudo systemctl start ntpd.service && \
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    grubby --args="transparent_hugepage=never" --update-kernel /boot/vmlinuz-3.10.0-957.el7.x86_64
+    ```
+    
+    > **注意：**
+    >
+    > --update-kernel 后需要使用实际的默认内核版本。
```
Comment (TomShawn) [inline check-before-deployment.md:276]:
```suggestion
    > `--update-kernel` 后需要使用实际的默认内核版本。
```

File: check-before-deployment.md [hunk 6]
```diff
@@ -229,6 +229,103 @@ sudo systemctl start ntpd.service && \
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    grubby --info /boot/vmlinuz-3.10.0-957.el7.x86_64
+    ```
+    
+    > **注意：**
+    >
+    > --info 后需要使用实际的默认内核版本。
```
Comment (TomShawn) [inline check-before-deployment.md:288]:
```suggestion
    > `--info` 后需要使用实际的默认内核版本。
```

File: check-before-deployment.md [hunk 7]
```diff
@@ -229,6 +229,103 @@ sudo systemctl start ntpd.service && \
...
+        ```
+
+    - 如果不希望重启机器，也可以修改当前的内核配置来立即生效：
+
+       {{< copyable "shell-regular" >}}
+
+        ```bash
+        echo never > /sys/kernel/mm/transparent_hugepage/enabled
+        echo never > /sys/kernel/mm/transparent_hugepage/defrag
+        ```
```
Comment (TomShawn) [inline check-before-deployment.md:315]:
```suggestion
    - 如果需要重启验证，执行 `reboot` 命令：

        {{< copyable "shell-regular" >}}

        ```bash
        reboot
        ```

    - 如果不希望重启机器，也可以修改当前的内核配置来立即生效：

        {{< copyable "shell-regular" >}}

        ```bash
        echo never > /sys/kernel/mm/transparent_hugepage/enabled
        echo never > /sys/kernel/mm/transparent_hugepage/defrag
        ```
```

File: check-before-deployment.md [hunk 8]
```diff
@@ -229,6 +229,103 @@ sudo systemctl start ntpd.service && \
...
+
+6. 查看重启或者修改后的默认内核生效配置，如果输出 `always madvise [never]` 表示透明大页处于禁用状态
```
Comment (TomShawn) [inline check-before-deployment.md:317]:
```suggestion
6. 查看重启或者修改后的默认内核生效配置。如果输出 `always madvise [never]` 表示透明大页处于禁用状态。
```

**Review by TomShawn — COMMENTED** [review]
minor comments on format

**Review by pepezzzz — COMMENTED** [review]
add tuned section

**Comment by pepezzzz** [comment]
@ethercflow PTAL

**Review by ethercflow — APPROVED** [review]
LGTM

File: check-before-deployment.md [hunk 9]
```diff
@@ -229,6 +229,143 @@ sudo systemctl start ntpd.service && \
 sudo systemctl enable ntpd.service
 ```
 
+## 检测及关闭透明大页
+
+对于数据库应用，不推荐使用 THP（transparent hugepage，透明大页），因为数据库往往具有稀疏而不是连续的内存访问模式，且当高阶内存碎片化比较严重时，分配 THP 页面会出现较大的延迟。若开启针对 THP 的直接内存规整功能，也会出现系统 CPU 使用率激增的现象，因此建议关闭 THP。
```
Comment (TomShawn) [inline check-before-deployment.md:234]:
```suggestion
对于数据库应用，不推荐使用透明大页（THP，即 transparent hugepage），因为数据库的内存访问模式往往是稀疏的而非连续的，且当高阶内存碎片化比较严重时，分配 THP 页面会出现较大的延迟。若开启针对 THP 的直接内存规整功能，也会出现系统 CPU 使用率激增的现象，因此建议关闭透明大页。
```

File: check-before-deployment.md [hunk 10]
```diff
@@ -229,6 +229,143 @@ sudo systemctl start ntpd.service && \
...
+    - 如果不希望重启机器，也可以修改当前的内核配置来立即生效：
+
+        {{< copyable "shell-regular" >}}
+
+        ```bash
+        echo never > /sys/kernel/mm/transparent_hugepage/enabled
+        echo never > /sys/kernel/mm/transparent_hugepage/defrag
+        ```
+
+6. 查看重启或者修改后的默认内核生效配置。如果输出 `always madvise [never]` 表示透明大页处于禁用状态。
```
Comment (TomShawn) [inline check-before-deployment.md:318]:
```suggestion
6. 查看重启或者修改后已生效的默认内核配置。如果输出 `always madvise [never]` 表示透明大页处于禁用状态。
```

File: check-before-deployment.md [hunk 11]
```diff
@@ -229,6 +229,143 @@ sudo systemctl start ntpd.service && \
...
+
+    ```bash
+    cat /sys/kernel/mm/transparent_hugepage/enabled
+    ```
+
+    ```
+    always madvise [never]
+    ```
+
+7. 如果透明大页禁用未生效，需要使用 tuned / ktune 动态内核调试工具修改透明大页内容配置。
```
Comment (TomShawn) [inline check-before-deployment.md:330]:
```suggestion
7. 如果透明大页禁用未生效，需要使用 tuned / ktune 动态内核调试工具修改透明大页内容配置。操作步骤如下：
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**